lib/util/src/logging.c

```diff
@@ -1007,8 +1022,110 @@ package_source_info(struct json_context *jc)
     json_element_field(jc, "hse_version", "%s", HSE_VERSION_STRING);
 }
 
+static bool
+slog_tab_process_fmt(const char *fmt, char *new_fmt, size_t size)
+{
+    char *s = strchr(fmt, '%');
+
+    if (!s)
+        return false;
+
+    switch (*(++s)) {
+        case 's':
+        case 'p':
+        case 'x':
+        case 'X':
+            snprintf(new_fmt, size, "%s ", fmt);
+            return true;
+    }
+
+    snprintf(new_fmt, size, "%s ", fmt);
+    return true;
+}
+
+#define slog_tab_snprintf(pc, ...) \
+    (snprintf_append(pc->buf, pc->buf_sz, &pc->off, __VA_ARGS__))
+
+#define slog_tab_vsnprintf(pc, ...) \
+    (vsnprintf_append(pc->buf, pc->buf_sz, &pc->off, __VA_ARGS__))
+
+static
+void
+slog_tab_element_fieldv(struct slog_tab_context *pc, va_list fields)
+{
+    char *key, *fmt;
+    char  new_fmt[32];
+
+    key = va_arg(fields, char *);
+    fmt = va_arg(fields, char *);
+
+    if (!slog_tab_process_fmt(fmt, new_fmt, sizeof(new_fmt)))
+        return;
+
+    slog_tab_snprintf(pc, "%s ", key);
+    slog_tab_vsnprintf(pc, new_fmt, fields);
+}
```

Comment:
I don't think I've ever seen any code that called va_arg() twice on the same argument....

Reply:
n/m, i was thinking of va_start()...

Reply:
I copied and edited from identical code on the json ouput path without much thought about what it was doing...